pkg/cmd/ssh/options.go

```diff
@@ -413,29 +414,37 @@ func (o *SSHOptions) Run(f util.Factory) error {
 		return err
 	}
 
-	// validate the current target
-	currentTarget, err := manager.CurrentTarget()
+	// sshTarget is the target used for the run method
+	sshTarget, err := manager.CurrentTarget()
 	if err != nil {
 		return err
 	}
 
-	if currentTarget.ShootName() == "" {
-		return errors.New("no Shoot cluster targeted")
-	}
-
-	printTargetInformation(o.IOStreams.Out, currentTarget)
-
 	// create client for the garden cluster
-	gardenClient, err := manager.GardenClient(currentTarget.GardenName())
+	gardenClient, err := manager.GardenClient(sshTarget.GardenName())
 	if err != nil {
 		return err
 	}
 
+	if sshTarget.ShootName() == "" {
+		if sshTarget.SeedName() != "" {
+			if shoot, err := gardenClient.GetShootOfManagedSeed(f.Context(), sshTarget.SeedName()); err != nil {
+				return err
+			} else if shoot != nil {
+				sshTarget = sshTarget.WithProjectName("garden").WithShootName(shoot.Name)
+			} else {
+				return target.ErrNoShootTargeted
+			}
+		}
+	}
```

Comment:
`if sshTarget.ShootName() == "" && sshTarget.SeedName() != "" {`
```suggestion
	if sshTarget.ShootName() == "" && sshTarget.SeedName() != "" {
		if shoot, err := gardenClient.GetShootOfManagedSeed(f.Context(), sshTarget.SeedName()); err != nil {
			return err
		} else if shoot != nil {
			sshTarget = sshTarget.WithProjectName("garden").WithShootName(shoot.Name)
		} else {
			return target.ErrNoShootTargeted
		}
	}
```